javascript/ql/src/Security/CWE-730/UnclosedStream.ql

```diff
@@ -56,6 +56,36 @@ class PromiseExecutor extends AsyncTerminatableFunction {
   override string getKind() { result = "promise" }
 }
 
+/**
+ * A callback-invoking function heuristic as a function that can be terminated in an asynchronous context.
+ */
+class FunctionWithCallback extends AsyncTerminatableFunction {
+  DataFlow::ParameterNode callbackParameter;
+
+  FunctionWithCallback() {
+    // the last parameter is the callback
+    callbackParameter = this.getLastParameter() and
+    // simple escape analysis
+    not exists(DataFlow::Node escape | callbackParameter.flowsTo(escape) |
+      escape = any(DataFlow::PropWrite w).getRhs() or
+      escape = any(DataFlow::CallNode c).getAnArgument()
+    ) and
+    // no return value
+    (this.getFunction() instanceof ArrowFunctionExpr or not exists(this.getAReturn())) and
```

Comment:
We could exclude arrow functions with an explicit return, like `() => { ...; return x }`
```suggestion
    (this.getFunction().getBody() instanceof Expr or not exists(this.getAReturn())) and
```